src/Utilities/CharString.f90

```diff
@@ -100,4 +101,11 @@ subroutine write_unformatted(this, unit, iostat, iomsg)
     end if
   end subroutine write_unformatted
 
+  function strlen(this) result(length)
+    class(CharacterStringType), intent(in) :: this
+    integer :: length
+
```

Comment:
should we say:

```
if (allocated(this%charstring)) then
  length = len(this%charstring))
else
  length = 0
end if
```

Reply:
I was thinking len should report 0 if not allocated but should have verified.  I'll make the change.